tests/e2e/Services/Sites/SitesCustomServerTest.php

```diff
@@ -539,6 +539,75 @@ public function testAdapterDetectionStaticSPA(): void
         $this->cleanupSite($siteId);
     }
 
+    public function testSettingsForRollback(): void
+    {
+        $siteId = $this->setupSite([
+            'siteId' => ID::unique(),
+            'name' => 'Static site',
+            'framework' => 'other',
+            'buildRuntime' => 'node-22',
+            'outputDirectory' => '',
+            'buildCommand' => '',
+            'installCommand' => '',
+        ]);
+        $this->assertNotEmpty($siteId);
+
+        $site = $this->getSite($siteId);
+        $this->assertEquals('200', $site['headers']['status-code']);
+        $this->assertArrayHasKey('adapter', $site['body']);
+        $this->assertArrayHasKey('fallbackFile', $site['body']);
+        $this->assertEmpty($site['body']['adapter']);
+        $this->assertEmpty($site['body']['fallbackFile']);
+
+        $domain = $this->setupSiteDomain($siteId);
+        $this->assertNotEmpty($domain);
+
+        $deploymentId1 = $this->setupDeployment($siteId, [
+            'code' => $this->packageSite('static-single-file'),
+            'activate' => 'true'
+        ]);
+        $this->assertNotEmpty($deploymentId1);
+
+        $site = $this->getSite($siteId);
+        $this->assertEquals('200', $site['headers']['status-code']);
+        $this->assertEquals('static', $site['body']['adapter']);
+        $this->assertEquals('main.html', $site['body']['fallbackFile']);
+
+        $site = $this->updateSite([
+            'fallbackFile' => 'hello.html',
```

Comment:
Let's set it empty here instead.

Also we might need to change outputDirectory for ssr astro deployment to work properly